Alignment/OfflineValidation/interface/outTrends.h

```diff
@@ -0,0 +1,191 @@
+#ifndef OUTTRENDS_H_
+#define OUTTRENDS_H_
+
+#include "TArrow.h"
+#include "TAxis.h"
+#include "TCanvas.h"
+#include "TF1.h"
+#include "TFile.h"
+#include "TGaxis.h"
+#include "TGraph.h"
+#include "TGraphAsymmErrors.h"
+#include "TGraphErrors.h"
+#include "TH1F.h"
+#include "TH2F.h"
+#include "TLatex.h"
+#include "TLegend.h"
+#include "TList.h"
+#include "TMath.h"
+#include "TObjArray.h"
+#include "TObjString.h"
+#include "TPad.h"
+#include "TPaveText.h"
+#include "TProcPool.h"
+#include "TProfile.h"
+#include "TROOT.h"
+#include "TStyle.h"
+#include "TSystem.h"
+#include "TSystemDirectory.h"
+#include "TSystemFile.h"
+#include <TStopwatch.h>
+#include <algorithm>
+#include <bitset>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <sstream>
+#include <vector>
+
+
+/*!
+ * \def basically the y-values of a TGraph
+ */
+typedef std::map<TString, std::vector<double> > alignmentTrend;
+
+// auxilliary struct to store
+// histogram features
+struct unrolledHisto {
+  double m_y_min;
+  double m_y_max;
+  unsigned int m_n_bins;
+  std::vector<double> m_bincontents;
+
+  unrolledHisto() {
+    m_y_min = 0.;
+    m_y_max = 0.;
+    m_n_bins = 0;
+    m_bincontents.clear();
+  }  // first constructor empty
+
+  unrolledHisto(const double &y_min,
+                const double &y_max,
+                const unsigned int &n_bins,
+                const std::vector<double> &bincontents) {
+    m_y_min = y_min;
+    m_y_max = y_max;
+    m_n_bins = n_bins;
+    m_bincontents = bincontents;
+  }  //look, a constructor
+
+  double get_y_min() { return m_y_min; }
+
+  double get_y_max() { return m_y_max; }
+
+  unsigned int get_n_bins() { return m_n_bins; }
+
+  std::vector<double> get_bin_contents() { return m_bincontents; }
+
+  double get_integral() {
+    double ret(0.);
+    for (const auto &binc : m_bincontents) {
+      ret += binc;
+    }
+    return ret;
+  }
+};
+
+
+struct outTrends {
```

Comment:
This is only for the PV, right? can we rename it sth like `outPVtrends` or sth like this?

Reply:
Done